src/DataDefinitionsBundle/Context/ContextFactoryInterface.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Data Definitions.
+ *
+ * LICENSE
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2016-2019 w-vision AG (https://www.w-vision.ch)
+ * @license    https://github.com/w-vision/DataDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ */
+
+declare(strict_types=1);
+
+namespace Wvision\Bundle\DataDefinitionsBundle\Context;
+
+use Pimcore\Model\DataObject\Concrete;
+use Wvision\Bundle\DataDefinitionsBundle\Model\DataDefinitionInterface;
+use Wvision\Bundle\DataDefinitionsBundle\Model\ExportDefinitionInterface;
+use Wvision\Bundle\DataDefinitionsBundle\Model\ExportMapping;
+use Wvision\Bundle\DataDefinitionsBundle\Model\ImportDefinitionInterface;
+use Wvision\Bundle\DataDefinitionsBundle\Model\ImportMapping;
+use Wvision\Bundle\DataDefinitionsBundle\Model\MappingInterface;
+use Wvision\Bundle\DataDefinitionsBundle\Provider\ImportDataSetInterface;
+
+interface ContextFactoryInterface
+{
+    public function createFetcherContext(
+        ExportDefinitionInterface $definition,
+        array $params
+    ): FetcherContextInterface;
+
+    public function createLoaderContext(
+        ImportDefinitionInterface $definition,
+        array $params,
+        array $dataRow,
+        ImportDataSetInterface $dataSet,
+        string $class
+    ): LoaderContextInterface;
+
+    public function createFilterContext(
+        DataDefinitionInterface $definition,
+        array $params,
+        array $dataRow,
+        ImportDataSetInterface $dataSet,
+        Concrete $object
+    ): FilterContextInterface;
+
+    public function createGetterContext(
+        DataDefinitionInterface $definition,
+        array $params,
+        Concrete $object,
+        ExportMapping $mapping,
+    ): GetterContextInterface;
+
+    public function createSetterContext(
+        DataDefinitionInterface $definition,
+        array $params,
+        Concrete $object,
+        ImportMapping $mapping,
+        array $dataRow,
+        ImportDataSetInterface $dataSet,
+        mixed $value,
+    ): SetterContextInterface;
+
+    public function createInterpreterContext(
+        DataDefinitionInterface $definition,
+        array $params,
+        array $dataRow,
+        ?ImportDataSetInterface $dataSet,
```

Comment:
Why `Import` here since it's not import specific?

Reply:
exactly, it can be used for exports too.